### PR TITLE
simplify rendered

### PR DESCRIPTION
**Pull request by arnetheduck**
* remove cruft
* fix missing elif cond parenthesis

### CODE REVIEW

File: compiler/renderer.nim [hunk 1]
```diff
@@ -1045,7 +1026,7 @@ proc infixArgument(g: var TSrcGen, n: PNode, i: int) =
           needsParenthesis = true
   if needsParenthesis:
     put(g, tkParLe, "(")
-  gsub(g, n, i)
+  gsub(g, n[i])
```
Comment (Araq) [inline compiler/renderer.nim:1029]: Makes it less robust. Keep in mind that we also render severely broken ASTs with this code.

File: compiler/renderer.nim [hunk 2]
```diff
@@ -979,12 +966,6 @@ proc doParamsAux(g: var TSrcGen, params: PNode) =
     putWithSpace(g, tkOpr, "->")
     gsub(g, params[0])
 
-proc gsub(g: var TSrcGen; n: PNode; i: int) =
-  if i < n.len:
-    gsub(g, n[i])
-  else:
-    put(g, tkOpr, "<<" & $i & "th child missing for " & $n.kind & " >>")
```
Comment (Araq) [inline compiler/renderer.nim:986]: Should be kept.

File: compiler/renderer.nim (hunk 1 above)
Comment (arnetheduck) [reply compiler/renderer.nim:1029]: well, the code currently uses half in one style hand half in the other, a bit randomly - this merely unifies it behind a single style. the `n[i]` style is also used for all other helpers starting with `lsub` so while you may have hoped this was more robust, the code will likely crashed before it get this far

Comment (Araq) [reply compiler/renderer.nim:1029]: Well so use the robust version consistently and also use an `lsub` with this check.

Comment (arnetheduck) [reply compiler/renderer.nim:1029]: ok - that's beyond the scope of what I can backport from nph which isn't affected by this problem - it'll have to stay like this for another day

**Pull request closed without merging**